### ISSUE TITLE
False positive "Explicit lifetimes given in parameter types where they could be elided"

### ISSUE THREAD

**Issue by boohba** [intellij-rust/intellij-rust#9210]
## Environment

* **IntelliJ Rust plugin version:** 0.4.176.4815-222
* **Rust toolchain version:** 1.65.0-nightly
* **IDE name and version:** IntelliJ IDEA Ultimate 2022.2.1
* **Operating system:** Arch Linux

## Problem description
False positive "Explicit lifetimes given in parameter types where they could be elided"

## Steps to reproduce
```rust
use log::debug;
use tokio::io::AsyncReadExt;
use tokio::net::TcpStream;
use tokio::sync::MutexGuard;
use crate::connection::Connection;

pub async fn handle_handshake<'a>(length: u32, mut socket: MutexGuard<'a, TcpStream>, connection: &mut Connection) {
    const HANDSHAKE_MAX_LENGTH: u32 = 1029;

    if length > HANDSHAKE_MAX_LENGTH {
        debug!("Handshake packet is too big: {}", length);

        return;
    }

    let mut buf = Vec::with_capacity(length as usize);

    if let Err(e) = socket.read_exact(&mut buf).await {
        debug!("Failed to read handshake packet: {}", e);

        return;
    }

    // TODO
}
```
![image](https://user-images.githubusercontent.com/58115165/185466439-850e708d-922b-4c9a-8d29-c4608e4fb853.png)
![image](https://user-images.githubusercontent.com/58115165/185466691-f90dafd6-8c72-4b9e-896e-2fcd72155720.png)
